R/zchunk_LA144.Commercial.R

```diff
@@ -54,56 +74,453 @@ module_gcam.usa_LA144.Commercial_DISABLED <- function(command, ...) {
+      left_join_error_no_match(L143.share_state_Pop_HDD_sR9, by = c("state", "subregion9", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(state, subregion9, fuel, service, year, value)
+
+    # Expand L144.in_EJ_sR9_comm_F_U_Y cooling to all states by multiplying by CDD share within subregion9
+    L144.in_EJ_state_comm_F_cooling_Y <- states_subregions %>%
+      select(state, subregion9) %>%
+      repeat_add_columns(tidyr::crossing(fuel = CBECS_cooling_fuels, year = HISTORICAL_YEARS)) %>%
+      mutate(service = "comm cooling") %>%
+      left_join_error_no_match(L144.in_EJ_sR9_comm_F_U_Y, by = c("subregion9", "fuel", "service", "year")) %>%
+      left_join_error_no_match(L143.share_state_Pop_CDD_sR9, by = c("state", "subregion9", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(state, subregion9, fuel, service, year, value)
+
+    # Downscaling all remaining services to states according to floorspace
+    # Calculate the share of each state within its census division's (subregion9's) floorspace
+    L144.flsp_state_share_sR9 <- L144.flsp_bm2_state_comm %>%
+      group_by(subregion9, year) %>%
+      mutate(share = value / sum(value)) %>%
+      ungroup() %>%
+      select(-value)
+
+    # Remove heating and cooling
+    L144.in_EJ_sR9_comm_F_Uoth_Y <- L144.in_EJ_sR9_comm_F_U_Y %>%
+      filter(!(service %in% c("comm heating", "comm cooling")))
+
+    # Convert to state data by multiplying by floorspace share
+    L144.in_EJ_state_comm_F_Uoth_Y <- L144.in_EJ_sR9_comm_F_Uoth_Y %>%
+      select(fuel, service) %>%
+      distinct() %>%
+      repeat_add_columns(tidyr::crossing(state = gcamusa.STATES, year = HISTORICAL_YEARS)) %>%
+      # We now have all combos of fuel, service, state and historical year; add in subregion9
+      left_join_error_no_match(states_subregions, by = "state") %>%
+      # Add in floorspace data
+      left_join_error_no_match(L144.flsp_state_share_sR9, by = c("state", "year", "subregion9")) %>%
+      # Add in energy data
+      left_join_error_no_match(L144.in_EJ_sR9_comm_F_Uoth_Y,
+                               by = c("subregion9", "fuel", "service", "year")) %>%
+      mutate(value = value * share) %>%
+      select(state, subregion9, fuel, service, year, value)
+
+    # Assembling unscaled energy consumption by state, fuel, and service
+    L144.in_EJ_state_comm_F_U_Y_unscaled <- bind_rows(L144.in_EJ_state_comm_F_heating_Y,
+                                                      L144.in_EJ_state_comm_F_cooling_Y,
+                                                      L144.in_EJ_state_comm_F_Uoth_Y)
+
+    # Calculating shares of energy consumption by each service, within each state and fuel
+    L144.pct_state_comm_F_U_Y <- L144.in_EJ_state_comm_F_U_Y_unscaled %>%
+      group_by(state, fuel, year) %>%
+      mutate(value = value / sum(value),
+             sector = "comm") %>%
```

Comment:
If this grouped mutate takes any nontrivial amount of time, it's probably faster to do this `sector = "comm"` assignment after the `ungroup`, i.e. do it once for the entire dataset, not within each individual group. Same comment applies to line 391 below.